From a9c0a44002997c9c708460c1a4a900365f5d84af Mon Sep 17 00:00:00 2001
From: Tom Zanussi <tom.zanussi@intel.com>
Date: Tue, 10 Jul 2012 09:37:56 -0500
Subject: [PATCH] perf: add sgidefs.h to for mips builds

Allow Python.h to find the definitions it needs on mips i.e. get rid
of the error: "_ABIN32" is not defined.

Signed-off-by: Tom Zanussi <tom.zanussi@intel.com>
---
 tools/perf/scripts/python/Perf-Trace-Util/Context.c    | 4 ++++
 tools/perf/util/scripting-engines/trace-event-python.c | 4 ++++
 2 files changed, 8 insertions(+)

diff --git a/tools/perf/scripts/python/Perf-Trace-Util/Context.c b/tools/perf/scripts/python/Perf-Trace-Util/Context.c
index 01f54d6724a5..3b6e975949b9 100644
--- a/tools/perf/scripts/python/Perf-Trace-Util/Context.c
+++ b/tools/perf/scripts/python/Perf-Trace-Util/Context.c
@@ -11,6 +11,10 @@
  */
 #define PY_SSIZE_T_CLEAN
 
+#ifdef __mips__
+#include <sgidefs.h>
+#endif
+
 #include <Python.h>
 #include "../../../util/config.h"
 #include "../../../util/trace-event.h"
diff --git a/tools/perf/util/scripting-engines/trace-event-python.c b/tools/perf/util/scripting-engines/trace-event-python.c
index 8bdae066e839..2eb0dae791b5 100644
--- a/tools/perf/util/scripting-engines/trace-event-python.c
+++ b/tools/perf/util/scripting-engines/trace-event-python.c
@@ -19,6 +19,10 @@
  *
  */
 
+#ifdef __mips__
+#include <sgidefs.h>
+#endif
+
 #include <Python.h>
 
 #include <inttypes.h>
-- 
2.39.2

